### PR TITLE
Error pages

### PR DESCRIPTION
**Pull request by okolomiets**

### CODE REVIEW

File: src/errorPage/notFound.component.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    public controller: any;
+    public templateUrl: string;
+    public controllerAs: string;
+
+    constructor() {
+        this.bindings = {
+          config: '<'
+        };
+		this.controller = ErrorController;
+        this.templateUrl = './errorPage/notFound.partial.html';
```
Comment (freeman14) [inline src/errorPage/notFound.component.ts]: Use template instead and require method to import template

**Review by freeman14 — CHANGES_REQUESTED** [review]

File: sb/.webpack/webpack.dev.js [hunk 2]
```diff
@@ -32,7 +33,15 @@ module.exports = {
             { from: './sb/.static/preview.html', to: 'preview.html' },
             { from: './sb/.static/index.html', to: 'index.html' }
         ]),
-        new webpack.HotModuleReplacementPlugin()
+        new webpack.HotModuleReplacementPlugin(),
+        new sprite({
+          'source' : path.resolve(__dirname, '..', '..', 'src/assets/images/'),
```
Comment (freeman14) [inline sb/.webpack/webpack.dev.js]: use `path.resolve(__dirname, '../../src/assets/images/')` instead

File: src/errorPage/error.component.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    public controller: any;
+    public templateUrl: string;
+    public controllerAs: string;
+
+    constructor() {
+        this.bindings = {
+          config: '<'
+        };
+		this.controller = ErrorController;
+        this.templateUrl = './errorPage/error.partial.html';
```
Comment (freeman14) [inline src/errorPage/error.component.ts]: Same with `require`

File: src/errorPage/maintenance.component.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    public controller: any;
+    public templateUrl: string;
+    public controllerAs: string;
+
+    constructor() {
+        this.bindings = {
+          config: '<'
+        };
+		this.controller = ErrorController;
+        this.templateUrl = './errorPage/maintenance.partial.html';
```
Comment (freeman14) [inline src/errorPage/maintenance.component.ts]: Same with `require `

**Review by freeman14 — CHANGES_REQUESTED** [review]

**Comment by okolomiets** [comment]
Fixed

File: src/styles/components/error.less [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    margin-left: 20px;
+    cursor: pointer;
+  }
+
+  .error-page-header-link {
+    font-size: 16px;
+    font-weight: 600;
+    color: @submarine;
+    padding-right: 20px;
+    padding-left: 20px;
```
Comment (freeman14) [inline src/styles/components/error.less]: padding: 0 20px

File: src/styles/components/error.less [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    &:hover {
+      color: @dusty-gray;
+    }
+  }
+
+  *:active,
+  *:focus,
+  *.active {
+    outline: 0;
+    -webkit-box-shadow: none;
```
Comment (freeman14) [inline src/styles/components/error.less]: https://css-tricks.com/do-we-need-box-shadow-prefixes/

File: src/styles/components/error.less [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    text-align: center;
+  }
+
+  .error-page-message {
+    max-width: 530px;
+    margin-top: 10px;
+    margin-bottom: 40px;
```
Comment (freeman14) [inline src/styles/components/error.less]: margin: 10px 0 40px;

File: src/styles/components/error.less [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    margin-top: 20px;
+
+    .error-page-splitter {
+      display: inline-block;
+      width: 2px;
+      height: inherit;
+      width: 2px;
+      background-color: @gainsboro;
+      margin-left: 10px;
+      margin-right: 10px;
```
Comment (freeman14) [inline src/styles/components/error.less]: margin: 0 10px;

**Review by freeman14 — CHANGES_REQUESTED** [review]

**Comment by okolomiets** [comment]
Fixed.

**Pull request merged**